src/services/review/RepoCheckerService.ts

```diff
@@ -64,6 +69,65 @@ export default class RepoCheckerService {
     this.repoRepository = repoRepository
   }
 
+  isCurrentlyLocked(repo: string): ResultAsync<boolean, never> {
+    const logsFilePath = path.join(SITE_CHECKER_REPO_PATH, repo)
+    // create logs folder if it does not exist
+
+    if (!fs.existsSync(logsFilePath)) {
+      fs.mkdirSync(logsFilePath, { recursive: true })
+    }
+
+    // check if checker.lock exists
+    const lockFilePath = path.join(logsFilePath, "checker.lock")
+    return ResultAsync.fromPromise(
+      fs.promises.access(lockFilePath, fs.constants.F_OK),
+      () => okAsync(false)
+    )
+      .map(() => true)
+      .orElse(() => ok(false))
+  }
+
+  createLock(repo: string): ResultAsync<undefined, SiteCheckerError> {
+    // create a checker.lock file
+    const lockFilePath = path.join(SITE_CHECKER_REPO_PATH, repo, "checker.lock")
+    return ResultAsync.fromPromise(
+      fs.promises.writeFile(lockFilePath, "locked"),
+      (error) => new SiteCheckerError(`${error}`)
+    )
+      .andThen(() =>
+        ResultAsync.fromPromise(
+          this.git
+            .cwd({ path: SITE_CHECKER_REPO_PATH, root: false })
+            .add([`${path.join(repo, "checker.lock")}`])
+            .commit(`Create lock for ${repo}`)
+            .push(),
+          (error) => new SiteCheckerError(`${error}`)
+        )
+      )
+      .map(() => undefined)
```

Comment:
do we need this? i feel that returning the result of the operation might be more useful?

Reply:
uhm wait ah if not success we error out no? the info is not lost, it returns an error?

Reply:
`map` only ever operates on success values! you might be looking for `mapErr` here

Reply:
changed api for this